File: lab-resource-platform/src/main/java/com/labplatform/labresourceplatform/service/InstitutionService.java
package com.labplatform.labresourceplatform.service;

import com.labplatform.labresourceplatform.entity.Institution;
import com.labplatform.labresourceplatform.repository.InstitutionRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class InstitutionService {

    private final InstitutionRepository institutionRepository;
    public InstitutionService(InstitutionRepository institutionRepository){
        this.institutionRepository=institutionRepository;
    }

    public List<Institution> getAllInstitutions(){
        return institutionRepository.findAll();
    }

    public Institution getInstitutionById(Long id){
        return institutionRepository.findById(id)
                .orElseThrow(()->new RuntimeException("Institution not found with Id: "+id));
    }

    public Institution createInstitution(Institution institution){
        return institutionRepository.save(institution);
    }

    public Institution updateInstitution(Long  id, Institution updatedInstitution){

        Institution existing = getInstitutionById(id);

        if(updatedInstitution.getInstitutionName()!=null)
            existing.setInstitutionName(updatedInstitution.getInstitutionName());
        if(updatedInstitution.getAddress()!=null)
            existing.setAddress(updatedInstitution.getAddress());
        if(updatedInstitution.getContactPhone()!=null)
            existing.setContactPhone(updatedInstitution.getContactPhone());
        if(updatedInstitution.getContactEmail()!=null)
            existing.setContactEmail(updatedInstitution.getContactEmail());

        return institutionRepository.save(existing);
    }

    public void deleteInstitution(Long id){
        Institution institution = getInstitutionById(id);
        institutionRepository.delete(institution);
    }
}

File: frontend/src/services/institutionService.js
import axiosInstance from "../utils/axiosInstance";

const API = "/institutions";

const getAllInstitutions = async () => {
    const response = await axiosInstance.get(API);
    return response.data;
};

const createInstitution = async (institution) => {
    const response = await axiosInstance.post(API, institution);
    return response.data;
};

const updateInstitution = async (id, institution) => {
    const response = await axiosInstance.put(`${API}/${id}`, institution);
    return response.data;
};

const deleteInstitution = async (id) => {
    return axiosInstance.delete(`${API}/${id}`);
};

export default {
    getAllInstitutions,
    createInstitution,
    updateInstitution,
    deleteInstitution,
};